woodpecker-cli - runtime error: invalid memory address or nil pointer dereference with 2.4.x

### Component

cli

### Describe the bug

Since updating to 2.4.x woodpecker-cli runs into a runtime error:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x20 pc=0x1004c71a8]

goroutine 1 [running]:
go.woodpecker-ci.org/woodpecker/v2/cli/internal/config.Load(0x14000591580)
        go.woodpecker-ci.org/woodpecker/v2/cli/internal/config/config.go:53 +0x268
go.woodpecker-ci.org/woodpecker/v2/cli/common.Before(0x14000591580)
        go.woodpecker-ci.org/woodpecker/v2/cli/common/hooks.go:53 +0x70
github.com/urfave/cli/v2.(*Command).Run(0x14000597760, 0x14000591580, {0x1400014c000, 0xc, 0xc})
        github.com/urfave/cli/v2@v2.27.1/command.go:219 +0x3d0
github.com/urfave/cli/v2.(*App).RunContext(0x140001d6e00, {0x101e2cca0, 0x102d70660}, {0x1400014c000, 0xc, 0xc})
        github.com/urfave/cli/v2@v2.27.1/app.go:337 +0x514
github.com/urfave/cli/v2.(*App).Run(...)
        github.com/urfave/cli/v2@v2.27.1/app.go:311
main.main()
        go.woodpecker-ci.org/woodpecker/v2/cmd/cli/main.go:26 +0x48
```

Here is an example. I export the variables first

```
export WOODPECKER_PROJECT="test/project"
export WOODPECKER_SERVER="https://our.server"
export WOODPECKER_TOKEN="asdf"
```

Then i run any command, like `woodpecker-cli pipeline ls` and the error occurs.

### System Info

```shell
{"source":"https://github.com/woodpecker-ci/woodpecker","version":"2.4.1"}

➜  ~ woodpecker-cli --version
woodpecker-cli version 2.4.1
```


### Additional context

_No response_

### Validations

- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Checked that the bug isn't fixed in the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]

--- a/cli/common/hooks.go
+++ b/cli/common/hooks.go
@@ -57,7 +57,7 @@ func After(_ *cli.Context) error {
 	if waitForUpdateCheck != nil {
 		select {
 		case <-waitForUpdateCheck.Done():
-		// When the actual command already finished, we still wait 250ms for the update check to finish
+		// When the actual command already finished, we still wait 500ms for the update check to finish
 		case <-time.After(time.Millisecond * 500):
 			log.Debug().Msg("Update check stopped due to timeout")
 			cancelWaitForUpdate(errors.New("update check timeout"))

--- a/cli/internal/config/config.go
+++ b/cli/internal/config/config.go
@@ -30,9 +30,12 @@ func Load(c *cli.Context) error {
 		return err
 	}
 
-	if config == nil && !c.IsSet("server-url") && !c.IsSet("token") {
-		log.Info().Msg("The woodpecker-cli is not yet set up. Please run `woodpecker-cli setup`")
-		return errors.New("woodpecker-cli is not setup")
+	if config == nil {
+		config = &Config{
+			LogLevel:  "info",
+			ServerURL: c.String("server-url"),
+			Token:     c.String("token"),
+		}
 	}
 
 	if !c.IsSet("server") {
@@ -54,6 +57,11 @@ func Load(c *cli.Context) error {
 		if err != nil {
 			return err
 		}
+	}
+
+	if config.ServerURL == "" || config.Token == "" {
+		log.Info().Msg("The woodpecker-cli is not yet set up. Please run `woodpecker-cli setup` or provide the required environment variables / flags.")
+		return errors.New("woodpecker-cli is not configured")
 	}
 
 	return nil

